### PR TITLE
Configurable sparsify margin parameter for the ParticleListAction service.

### PR DESCRIPTION
**Pull request by drivera-fnal**

### CODE REVIEW

**Comment by absolution1** [comment]
trigger build

File: larg4/pluginActions/ParticleListAction_service.cxx [hunk 1]
```diff
@@ -77,7 +77,8 @@ namespace larg4 {
       fKeepEMShowerDaughters( p.get<bool>("keepEMShowerDaughters",true) ),
       fNotStoredPhysics( p.get< std::vector<std::string> >("NotStoredPhysics",{})),
       fkeepOnlyPrimaryFullTraj( p.get<bool>("keepOnlyPrimaryFullTrajectories",false) ),
-      fSparsifyTrajectories( p.get<bool>("SparsifyTrajectories",false) )
+      fSparsifyTrajectories( p.get<bool>("SparsifyTrajectories",false) ),
+      fSparsifyMargin( p.get<double>("SparsifyMargin",0.1) )
```
Comment (absolution1) [inline larg4/pluginActions/ParticleListAction_service.cxx]: Is a default parameter desired here?  
Not having one would make it easier to catch when the fcl param `SparsifyMargin` has been misconfigured.

Comment (drivera-fnal) [reply larg4/pluginActions/ParticleListAction_service.cxx]: Yes -- valid suggestion. I have removed the default.

**Comment by absolution1** [comment]
approve

**Comment by knoepfel** [comment]
@drivera-fnal, some DUNE CI tests are failing because they lack the new configuration parameter.  Should the configuration parameter be required, or can it be optional?

**Comment by absolution1** [comment]
Could the parameter be defined in the module configuration fcl (that's presumably within larsim)?

**Comment by drivera-fnal** [comment]
Hi Kyle,

The ProtoDUNE CI test should be updated to include the new parameter. If
the SparsifyTrajectories fcl parameter is set, then so should the new
SparsifyMargin parameter. I can reach out to Christoph Alt about this, or I
can make the change in dunetpc myself.

**Comment by lgarren** [comment]
@drivera-fnal   If changes are required in experiment code, please provide feature branches or PR's for all experiments that need these changes.

**Comment by drivera-fnal** [comment]
> @drivera-fnal If changes are required in experiment code, please provide feature branches or PR's for all experiments that need these changes.

@lgarren I added the new SparsifyMargin parameter to the ProtoDUNE-SP fhicl files and have pushed this to : `dunetpc` `feature/drivera_pdsp_sparsify_margin`

(No need to touch the CI files.)

**Comment by drivera-fnal** [comment]
> Could the parameter be defined in the module configuration fcl (that's presumably within larsim)?

@absolution1 Currently, I don't have a "general" configuration fhicl for the refactored Particle List Action service. I probably should create one and add it to larsim in a future PR. For now I just added the missing parameter to the PLA service configuration in the PDSP fhicls that needed it.

**Comment by lgarren** [comment]
The combined CI results are here.  Note that the CI is still in progress at this time.
https://dbweb8.fnal.gov:8443/LarCI/app/view_builds/index?offset=0&builds=lar_ci/10239&builds=

**Comment by lgarren** [comment]
CI tests are complete and normal.

**Comment by lgarren** [comment]
approve

**Pull request merged**